blueprints/layered-profile.md

```diff
@@ -0,0 +1,274 @@
+- `/servers/{id} PUT, DELETE`
+
+#### REST API Impact
+
+- No new endpoints are required
+
+**Existing Endpoints**
+
+- Modify JSON request and response for existing delivery services and servers endpoints.
+- JSON **response** with the proposed change will look as follows:
+
+`/deliveryservices?id=100`
+```JSON
+{ 
+  "id": 100,
+  ⋮
+  "requested": {
+    "profileIds": [
+      1234,
+      5678,
+      9012
+    ],
+    "profileNames": [
+      "EDGE",
+      "ATS8",
+      "TOP"
+    ]
+    ⋮
+  }
+}
+```
+
+`/servers?id=5`
+```JSON
+{ 
+  "id": 5,
+  ⋮
+  "requested": {
+    "profileIds": [
+      1357,
+      2468,
+      2356
+    ],
+    "profileNames": [
+      "MID",
+      "TOP",
+      "PRIMARY"
+    ],
+    ⋮
+  }
+}
+```
+
+JSON **request** with the proposed change will look as follows:
+
+`/deliveryservices`
+```JSON
+{
+  "requested": {
+    "profileIds": [
+      1234,
+      5678,
+      9012
+    ],
+    "profileNames": [
+      "EDGE",
+      "ATS8",
+      "TOP"
+    ],
+    ⋮
+  },
+  ⋮
+}
+```
+
+`/servers`
+```JSON
+{ 
+  "requested": {
+    "profileIds": [
+      1357, 
+      2468, 
+      2356
+    ],
+    "profileNames": [
+      "MID", 
+      "TOP", 
+      "PRIMARY"
+    ],
+    ⋮
+  },
+  ⋮
+}
+```
+
+The following table describes the top level `layered_profile` object for delivery services:
+
+| field              | type          | optionality | description                                                    |
+| ------------------ | --------------| ----------- | ---------------------------------------------------------------|
+| deliveryservice    | bigint        | required    | the delivery service id associated with a given profile        |
+| profile            | bigint        | required    | the profile id associated with a delivery service              |
+| order              | bigint        | required    | the order in which a profile is applied to a delivery service  |
+| lastUpdated        | bigint        | required    | the last time this delivery service was updated                |
+
+The following table describes the top level `layered_profile` object for servers:
+
+| field       | type                 | optionality | description                                                    |
+| ----------- | ---------------------| ----------- | ---------------------------------------------------------------|
+| server      | bigint               | required    | the server id associated with a given profile                  |
+| profile     | bigint               | required    | the profile id associated with a server                        |
+| order       | bigint               | required    | the order in which a profile is applied to a server            |
+| lastUpdated | bigint               | required    | the last time this server was updated                          |
+
+**API constraints**
+- TO REST APIs will not be backward compatibility and this feature will be a major version (v5.0) change.
```

Comment:
compatibility -> compatible

Also, this makes it sound like prior API versions won't be backwards-compatible, which they have to be. Maybe it should be worded like this:
> Prior API versions will continue to work as-is, with the profileId and profileName being the 1st profile in the list, whereas v5.0 will present the entire list of profile names.

Reply:
RE the conversation between Rob and I, that may not be the case, actually.

Reply:
Ah, I see that now.